### PR TITLE
Adds support for plugin params. 

### LINKED ISSUE #2203: Config management plugin environment variable UI/CLI support
**Is your feature request related to a problem? Please describe.**
Config management plugins support parameterizing using environment variables. Hoverwer it is impossible to configure env variable using CLI/UI

**Describe the solution you'd like**
User should be able to specify env variables using UI and CLI.

### PR DESCRIPTION
**Pull request by alexec**
Closes #2203

## View App Params
Screen Shot 2019-10-02 at 18 19 17

### New/Update App Params
Screen Shot 2019-10-03 at 13 21 53

### CODE REVIEW

**Comment by alexec** [comment]
@alexmt bit of a WIP. I really want you to look at `ArrayInputField` please

File: ui/src/app/applications/components/application-parameters/application-parameters.tsx [hunk 1]
```diff
@@ -199,6 +200,21 @@ export const ApplicationParameters = (props: {
...
+        attributes.push({
+            title: 'NAME',
+            view: app.spec.source.plugin && app.spec.source.plugin.name,
+            edit: (formApi: FormApi) => (
+                <FormField formApi={formApi} field='spec.source.plugin.name' component={Text}/>
+            ),
+        });
+        attributes.push({
+            title: 'ENV',
+            view: app.spec.source.plugin && (app.spec.source.plugin.env || []).map((i) => `${i.name}=${i.value}`).join(', '),
```
Comment (alexec) [inline ui/src/app/applications/components/application-parameters/application-parameters.tsx]: ugly code - refactor

File: ui/src/app/shared/components/array-input/array-input.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    public name: string;
+    public value: string;
+}
+
+class Props {
+    public setValue: (value: Item[]) => void;
+    public getValue: () => Item[];
+}
+
+class State {
```
Comment (alexec) [inline ui/src/app/shared/components/array-input/array-input.tsx]: either (a) replace `State` with `Item`, or (b) have a field in `State` called `newItem`

File: ui/src/app/shared/components/array-input/array-input.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                    </button>
+                </div>
+            </div>
+        );
+    }
+}
+
+export const ArrayInputField = ReactForm.FormField((props: { fieldApi: ReactForm.FieldApi }) => {
+    const {fieldApi: {getValue, setValue}} = props;
+    return <ArrayInput getValue={getValue} setValue={setValue}/>;
```
Comment (alexec) [inline ui/src/app/shared/components/array-input/array-input.tsx]: abstraction could be improved here but having some properties that are `replaceItem` `addItem` and `removeItem` callbacks

File: server/settings/settings.go [hunk 4]
```diff
@@ -27,6 +27,11 @@ func (s *Server) Get(ctx context.Context, q *settingspkg.SettingsQuery) (*settin
 	if err != nil {
 		return nil, err
 	}
+	overrides := make(map[string]*v1alpha1.ResourceOverride)
+	for k := range resourceOverrides {
+		val := resourceOverrides[k]
+		overrides[k] = &val
+	}
```
Comment (alexec) [inline server/settings/settings.go]: lift and shift

**Review by alexmt — APPROVED** [review]
LGTM. Please added spaces between textboxes and `=` and `+`, `x` icons.

**Pull request merged**